libs/@guardian/source-foundations/src/utils/convert-value.ts

```diff
@@ -1,5 +1,7 @@
 export const rootPixelFontSize = 16;
 export const pxToRem = (px: number): number => px / rootPixelFontSize;
 export const pxStringToNumber = (px: string): number => Number(px.slice(0, -2));
-export const fontArrayToString = (fonts: string[]): string =>
+export const fontArrayToString = (
+	fonts: string[] | readonly string[],
+): string =>
```

Comment:
A mutable string is always acceptable where a readonly one can be passed, so we only need to make the function signature accept immutable ones:

```suggestion
export const fontArrayToString = (
	fonts: readonly string[],
): string =>
```

[TS Playground](https://www.typescriptlang.org/play?target=99#code/MYewdgziA2CmB0w4EMBOAKAlAbgFC9EgBcACAWwFcjkAjOEgXhIG0BGAGhICZOBmTgCycArJwBsAXTyEIpAJZlK1OrEYsO3PoJHiJJZBBIyieAuFkkADqhDBYEQ03RpUyAJ4AuEqljIAJuDQbiRgFGQ0sKjMEpiMAHz6qK5u8GTIluih4ZHxIWERqPBEIADKRKhyYADm6FyYmPgyMAjQIDXWtvYQ6Eq0cPVmkM3wre02dg7oCr0q9UA)

Reply:
that doesn't work when it's a readonly prop:

<img width="434" alt="image" src="https://github.com/guardian/csnx/assets/867233/afedb7dd-0a99-46ad-8ce2-30cf75d26373">

[TS Playgorund](https://www.typescriptlang.org/play?target=99#code/MYewdgziA2CmB0w4EMBOAKAlAbgFC9EgBcACADxIF4SBvXEhkgTwC4SBtAcgAsBLTgLq4AviWQQShCEXwAzAK5hgRXuBIAvdMjbTUvMAHN2AzLRLD8msvCaYgA)

Reply:
So my point is that the method `fontArrayToString` never relies on the fact that the `fonts` array it receives is mutable, so it’s safe to update its signature to only consider a `readonly string[]`. This is because all mutable arrays can be provided where an immutable one is required, but not the other way around.

[TS Playground](https://www.typescriptlang.org/play?target=99#code/MYewdgziA2CmB0w4EMBOAKAlAbgFC9EgBcACADxIF4SBvXEhkgTwC4SBtAcgAsBLTgLq4AviWQQShCEXxTSZAExVa9Rqw6cARk1iCANCLy4AZgFcwwIr3AkAXumRtUsZABNw0JiWmpeYAObsApi0JML49mTwTJi4kQrRmEA)